src/reactviews/pages/SchemaCompare/components/SchemaSelectorDrawer.tsx

```diff
@@ -62,11 +103,24 @@ const SchemaSelectorDrawer = (props: Props) => {
     const classes = useStyles();
 
     const context = useContext(schemaCompareContext);
-    const [schemaType, setSchemaType] = useState("database");
+
+    const currentEndpoint =
+        props.endpointType === "source"
+            ? context.state.sourceEndpointInfo
+            : context.state.targetEndpointInfo;
+
+    const [schemaType, setSchemaType] = useState(
+        endpointTypeToString(currentEndpoint?.endpointType) || "database",
```

Comment:
Avoid duplicating a string literal by putting in inside the call

```suggestion
        endpointTypeToString(currentEndpoint?.endpointType || SchemaCompareEndpointType.Database),
```